test/Quaternion.jl

```diff
@@ -522,7 +522,13 @@ Base.:(/)(a::MyReal, b::Real) = a.val / b
             for _ in 1:100
                 q1 = nquatrand()
                 q2 = qrotation(rotationmatrix(q1), q1)
+                q3 = qrotation(rotationmatrix(q1))
                 @test q1 ≈ q2
+                @test q2 === q3 || q2 === -q3
+                @test real(q3) ≥ 0
```

Comment:
Have you confirmed that this fails before the changes in this PR?

Reply:
Yes, the test failed on the current master branch.